integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/badrecordloger/BadRecordEmptyDataTest.scala

```diff
@@ -101,6 +101,47 @@ class BadRecordEmptyDataTest extends QueryTest with BeforeAndAfterAll {
     }
   }
 
+  test("Test complex type with empty values and IS_EMPTY_DATA_BAD_RECORD property") {
+    sql("DROP TABLE IF EXISTS complexcarbontable")
+    sql("DROP TABLE IF EXISTS complexhivetable")
+    sql(
+      "create table complexcarbontable(deviceInformationId int, channelsId string, ROMSize " +
+      "string, ROMName String, purchasedate string, mobile struct<imei:string, imsi:int>, MAC " +
+      "array<string>, locationinfo array<struct<ActiveAreaId:int, ActiveCountry:string, " +
+      "ActiveProvince:string, Activecity:string, ActiveDistrict:string, ActiveStreet:string>>, " +
+      "proddate struct<productionDate:string,activeDeactivedate:array<string>>, gamePointId " +
+      "double,contractNumber double)  STORED AS carbondata")
+    sql("LOAD DATA local inpath '" + resourcesPath +
+      "/complextypeWithEmptyRecords.csv' INTO table complexcarbontable OPTIONS('DELIMITER'=','," +
+      "'QUOTECHAR'='\"', 'FILEHEADER'='deviceInformationId,channelsId,ROMSize,ROMName," +
+      "purchasedate,mobile,MAC,locationinfo,proddate,gamePointId,contractNumber', " +
+      "'COMPLEX_DELIMITER_LEVEL_1'='$', 'COMPLEX_DELIMITER_LEVEL_2'=':', " +
+      "'bad_records_logger_enable'='true','IS_EMPTY_DATA_BAD_RECORD'='true' ," +
+      "'bad_records_action'='ignore')")
```

Comment:
can you add one validation with 'IS_EMPTY_DATA_BAD_RECORD'='true' and 'bad_records_action'='fail'